Source/DafnyCore/AST/TypeDeclarations/Declaration.cs

```diff
@@ -134,7 +137,7 @@ public override string ToString() {
 
   // For Boogie
   internal VerificationIdGenerator IdGenerator = new();
-  public override IEnumerable<INode> Children => (Attributes != null ? new List<Node> { Attributes } : Enumerable.Empty<Node>());
+  public override IEnumerable<INode> Children => Enumerable.Empty<INode>(); // Attributes should be enumerated by the parent, as they could be placed in different places
```

Comment:
Attributes are a chained list such that `{:a} {:b} {:c}` is converted into `Attribute{Name="c", Prev=Attribute{Name="b", Prev=Attribute{Name = "a", Prev = null } } }`
As such, it made sense and was acceptable that the last attribute `{:c}` is one of the children nodes of the declaration, and is itself a parent of `{:b}` which is itself a parent of `{:a}`. As such, the Start of both `{:c}` and `{:b}` are `{:a}`, which is ok but weird.
Now, these new @-attributes are stored in the same linked list, except they are placed usually a few keywords away, like 
`@IsolateAssertion method {:oldATtribute} Test()`.
so that would mean that the last attribute's span, `{:oldAttribute}`, starts at `@IsolateAssertion`. This is not acceptable for formatting as every token in a span not covered by a child belongs to the node, so the attribute would capture `method`. That is not acceptable. Moreover, now every attribute is positioned differently depending on its syntax.

With this change, attributes are no longer a single child under a declaration, but each attribute becomes an indpendent child, so that there is no conflict at all.